api/protos/bootstrap/v1/bootstrap.proto

```diff
@@ -35,10 +34,12 @@ message Server {
     SocketAddress address = 1 [(validate.rules).message.required = true];
 }
 
-// [#next-free-field: 2]
+// [#next-free-field: 3]
 message Upstream {
     // The address for the upstream cluster.
     SocketAddress address = 1 [(validate.rules).message.required = true];
+
+    string keep_alive_time = 2;
```

Comment:
can you leave a comment explaining what this is and why its type is `string`?

Reply:
I`ve noticed a usual way of representing time in human readable format. I can include examples in comment.

Reply:
the fact that the type is string makes me a bit concerned since it has the potential for being misused. How much trouble would it be to make it a `google.protobuf.Duration`?

Reply:
Changing it to `google.protobuf.Duration` is straight forward, but we keepalive will be in minutes or possibly hours and Duration does not give us a good human readable representation apart from strings
From https://developers.google.com/protocol-buffers/docs/reference/java/com/google/protobuf/Duration 

```
In JSON format, the Duration type is encoded as a string rather than an
 object, where the string ends in the suffix "s" (indicating seconds) and
 is preceded by the number of seconds, with nanoseconds expressed as
 fractional seconds. For example, 3 seconds with 0 nanoseconds should be
 encoded in JSON format as "3s", while 3 seconds and 1 nanosecond should
 be expressed in JSON format as "3.000000001s", and 3 seconds and 1
 microsecond should be expressed in JSON format as "3.000001s".
```